### ISSUE TITLE
[1.7.10-4.2.16.64] Letter exploit

### ISSUE THREAD

**Issue by Antifluxfield** [ForestryMC/ForestryMC#1586]
Letters can be used for copying items:
Step 1: Take a stack of empty letters with your inventory full. The letters should at least be opened once, so that their nbt won't change when you open them again.
Step 2: Drop the letters on the ground with only one of them remaining.
Step 3: Open the remaining letter, put in the items that you want to copy. During this process, make sure to sustain your inventory full, so you won't pick up the letters on the ground in this step.
Step 4: Close the letter GUI. At this moment, the empty letters on the ground will be pick up and stack with the edited letter in your hand, so the items in the letter will be duplicated in each of them, up to 64 times.

**Comment by mezz** [comment]
I followed your instructions but was unable to duplicate items, the results of step 4 do not happen because the letters do not stack.
Once the letters have been opened, they get a unique ID in NBT so they will not stack with each other. Only one letter can be opened at a time, so they get different unique IDs.

Please let me know if you have further steps I can follow. 
Please try to replicate this with only Forestry and no other mods, it could involve a second mod.

**Comment by Antifluxfield** [comment]
Cannot replicate under pure Forestry, survival.
Tried to reproduce and analyse the process, and seems this bug is not that severe.
The major cause is that I copied the empty letters by using Ctrl+left-click in creative mode with NEI, so these letters are stackable since they have the same UID; or as I discovered it at first, the letter is provided by IC2 trade-o-mat in a server, set by the op.
By the way, the letters on the ground are picked up and stacked with the letter in your hand at the moment when I closed the editing GUI. Seems that the data of the items in the letter are saved after the GUI is closed, and the duplication happens between these two events.
